### PR TITLE
Enable waiting for timeslot during CSMA-CA procedure

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/nrf_802154_config.h [hunk 1]
```diff
@@ -452,14 +452,13 @@ extern "C" {
  * Indicates whether the CSMA-CA algorithm waits for an available timeslot before it performs the CCA procedure.
  *
  * @note When this option is enabled, the CSMA-CA procedure may be synchronized with timeslots of
- *       other protocols. It decreases robustness of the CSMA-CA procedure and is not recommended.
- *       This option may be used as a workaround to problems with higher layers that do not correctly handle
- *       multiple CHANNEL_BUSY errors in a row. This potentially problematic situation may
- *       happen during activity of other protocols.
+ *       other protocols. It decreases robustness of the CSMA-CA procedure. On the other hand it increases
```
Comment (greg-fer) [inline src/nrf_802154_config.h]:
```suggestion
 *       other protocols. This decreases the robustness of the CSMA-CA procedure. On the other hand, this increases

**Comment by hubertmis** [comment]
@wbober @LuDuda FYI
I'll merge it when all tests (including weekly) pass

File: src/nrf_802154_config.h [hunk 2]
```diff
@@ -452,14 +452,13 @@ extern "C" {
...
+ *       performance of multi-protocol applications, especially when other protocols than 802.15.4 utilizes
```
Comment (greg-fer) [inline src/nrf_802154_config.h]:
```suggestion
 *       the performance of multi-protocol applications, especially when protocols other than 802.15.4 use
```

**Comment by hubertmis** [comment]
Passed weekly: http://projecttools.nordicsemi.no/bamboo/browse/KNG-TDST-96

**Pull request merged**